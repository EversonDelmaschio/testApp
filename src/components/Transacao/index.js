import React from 'react';
import {Text, View, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';

function Transacao({data, tipo, valor}) {
  return (
    <View style={styles.container} testID="transacao-1">
      <View>
        <Text style={styles.label}>Tipo Movimen.</Text>
        <Text>{tipo}</Text>
      </View>
      <View>
        <Text style={styles.label}>Data</Text>
        <Text>{data}</Text>
      </View>
      <View>
        <Text style={styles.label}>Valor</Text>
        <Text style={styles.valor}>
          P$ {Number(valor).toFixed(2).replace('.', ',')}
        </Text>
      </View>
    </View>
  );
}

Transacao.defaultProps = {
  tipo: '',
  valor: '0',
  data: '',
};

Transacao.propTypes = {
  tipo: PropTypes.string,
  valor: PropTypes.string,
  data: PropTypes.string,
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 10,
    paddingHorizontal: 15,
    borderRadius: 6,
    backgroundColor: 'rgba(0, 111, 185, 0.2)',
  },
  label: {
    fontSize: 11,
    color: 'rgba(85, 85, 85, 1)',
    marginBottom: 4,
  },
  valor: {
    fontWeight: '600',
  },
});

export default Transacao;
