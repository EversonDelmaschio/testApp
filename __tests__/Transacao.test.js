import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import Transacao from '../src/components/Transacao';

describe('Component de Transacao do extrato', function () {
  it('Deve renderizar igual ao snapshot', function () {
    const {toJSON} = render(
      <Transacao data={'08/09/2022'} tipo={'saque'} valor={'20.00'} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
