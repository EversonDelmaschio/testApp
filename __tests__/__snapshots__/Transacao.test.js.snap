// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component de Transacao do extrato Deve renderizar igual ao snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(62,160,63,0.3)",
      "marginTop": 10,
      "width": "90%",
    }
  }
  testID="transacao-1"
>
  <Text>
    08/09/2022
  </Text>
  <Text>
    saque
  </Text>
  <Text>
    R$ 
    20.00
  </Text>
</View>
`;
